crates/router_env/src/logger/setup.rs

```diff
@@ -33,22 +32,19 @@ pub fn setup(
     config: &config::Log,
     service_name: &'static str,
     crates_to_filter: impl AsRef<[&'static str]>,
-) -> Result<TelemetryGuard, opentelemetry::metrics::MetricsError> {
+) -> TelemetryGuard {
     let mut guards = Vec::new();
 
     // Setup OpenTelemetry traces and metrics
-    let (telemetry_tracer, _metrics_controller) = if config.telemetry.enabled {
-        global::set_text_map_propagator(TraceContextPropagator::new());
-        (
-            setup_tracing_pipeline(&config.telemetry, service_name),
-            setup_metrics_pipeline(&config.telemetry),
-        )
+    let traces_layer = if config.telemetry.traces_enabled {
+        setup_tracing_pipeline(&config.telemetry, service_name)
     } else {
-        (None, None)
+        None
     };
-    let telemetry_layer = match telemetry_tracer {
-        Some(Ok(ref tracer)) => Some(tracing_opentelemetry::layer().with_tracer(tracer.clone())),
-        _ => None,
+    let _metrics_controller = if config.telemetry.metrics_enabled {
+        setup_metrics_pipeline(&config.telemetry)
+    } else {
+        None
```

Comment:
```suggestion
    let _metrics_controller = config
        .telemetry
        .metrics_enabled
        .then(|| setup_metrics_pipeline(&config.telemetry))
        .flatten();
```
optional change.

Reply:
I'd personally prefer keeping it the way it is since I find it to be more readable, and since the suggested change requires understanding what `bool::then()` does to understand what's happening here. I'm open to any more opinions on this, I'll leave this conversation open for a while.

Reply:
Sure. It's an optional change. I don't prefer it either but suggested because there were already similar functions(ex: `fp_utils::when`) used in the codebase. You don't have to make the change. Resolving conv.

Reply:
Marking this resolved.